### PR TITLE
Add Microsoft.DataCatalog

### PR DESCRIPTION
**Pull request by alanshi-ms**
Code review signed off by daschult.

### CODE REVIEW

File: schemas/2016-03-30/Microsoft.DataCatalog.json [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+              "description": "Microsoft.DataCatalog/catalogs: users - specifies the list of catalog users"
+            }
+          },
+          "required": [
+            "sku"
+          ]
+        }
+      },
+      "required": [
+        "type",
```
Comment (wenwu449) [inline schemas/2016-03-30/Microsoft.DataCatalog.json]: double check, is "location" required for catalogs? if so, please add it into "required" array (not need to define it in properties though)

File: schemas/2016-03-30/Microsoft.DataCatalog.json [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        "properties"
+      ],
+      "description": "Microsoft.DataCatalog/catalogs resource"
+    }
+  },
+  "definitions": {
+    "user": {
+      "type": "object",
+      "properties": {
+        "upn": {
```
Comment (wenwu449) [inline schemas/2016-03-30/Microsoft.DataCatalog.json]: none of the properties are required?

File: schemas/2016-03-30/Microsoft.DataCatalog.json (hunk 1 above)
Comment (alanshi-ms) [inline schemas/2016-03-30/Microsoft.DataCatalog.json]: added

File: schemas/2016-03-30/Microsoft.DataCatalog.json (hunk 2 above)
Comment (alanshi-ms) [inline schemas/2016-03-30/Microsoft.DataCatalog.json]: added

File: tests/2016-03-30/Microsoft.DataCatalog.tests.json [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+          "units": 1,
+          "enableAutomaticUnitAdjustment": false,
+          "admins": [{"upn" : "myupn@microsoft.com", "objectId" : "99999999-9999-9999-9999-999999999999"}], 
+          "users":  [{"upn" : "myupn@microsoft.com", "objectId" : "99999999-9999-9999-9999-999999999999"}] 
+        }
+      }
+    }
+  ]
+}
+Enter file contents here
```
Comment (wenwu449) [inline tests/2016-03-30/Microsoft.DataCatalog.tests.json]: typo? Please use "node runSchemaTests.js" to verify before commit.

**Pull request merged**